appscale/tools/remote_helper.py

```diff
@@ -940,41 +942,46 @@ def terminate_virtualized_cluster(cls, keyname, is_verbose):
 
     acc = AppControllerClient(shadow_host, secret)
     try:
-      all_ips = acc.get_all_public_ips()
+      terminate_thread = threading.Thread(target=acc.run_terminate,
+                                          args=(clean,))
+      terminate_thread.start()
+      terminated_successfully = True
+      log_dump = ""
+      while not acc.is_appscale_terminated():
```

Comment:
`run_with_timeout` will return the default string passed in upon timeout. Any non-empty string will evaluate as true here, which I'm not sure you want.